overlord/snapstate/snapstate_update_test.go

```diff
@@ -14127,8 +14475,8 @@ type updateWithComponentsOpts struct {
 }
 
 func componentNameToType(c *C, name string) snap.ComponentType {
-	typ := strings.TrimSuffix(name, "-component")
-	if typ == name {
+	typ, _, ok := strings.Cut(name, "-component")
```

Comment:
nitpicking, but this is not doing the same as TrimSuffix (not that I expect a component to be called xx-componentblah-component, but still...)

Reply:
I did it this way because I wanted to be able to add some components like "test-component-extra", the difference was intentional.

Reply:
Ok, I see, sounds good